extensions/antora/xref-hash-validator/xref-hash-validator.js

```
@@ -0,0 +1,142 @@
+                    return [id]
+                    
+                }
+                return []
+            })
+            
+            // any anchor element in the HTML that has an href attribute can originate from an xref in the source file for this HTML page
+            const anchors = parseHTML(file.contents.toString()).querySelectorAll('a')
+            file.xrefChecker.internalLinks = anchors.flatMap((a) => {
+                // ignore unresolved links that Antora has already logged
+                if (a.classList.contains('unresolved')) return []
+
+                const href = a.getAttribute('href')
+
+                if (!href) return []
+
+                // if an id on this page matches this href, we can ignore it - it is essentially self-verifying
+                if (file.xrefChecker.linkTargets.includes(href.replace(/^#/, ''))) return []
+
+                if (href && !href.startsWith('http') && !href.startsWith('/docs')) {
+                    return [href]
+                }
+                return []
+            })
+
+        })
+        
+        // we can go back into the contentCatalog and check the xref data we just added
+        // for every file in the contentCatalog, check if the xrefChecker.internalLinks are valid
+        // by verifying that for an internal link to a section, the hash does actually exist in the target page
+        files.forEach( (file) => {
+            if (!file.out || !file.asciidoc) return
+    
+            // what is the full url of each target?
+            file.xrefChecker.internalLinks.forEach((link) => {
+
+                searchForThisURL = new URL(path.join(file.pub.url, link), playbook.site.url)
+
+                // if there's no hash the xref is already checked by Antora
+                // this check shouldn't ever match anyway because we should have already filtered out links without hashes
+                if (!searchForThisURL.hash) return
```

Given that this condition should never verify, shall we raise a warning/error if it is? Otherwise it will silently hide our crap.